### ISSUE TITLE
Open "Reservierung.js" instead of html document

### ISSUE THREAD

**Issue by garritfra** [garritfra/SchulenbergOberharz#23]
Currently, the reservation form opens a js script embedded in html.
It should however execute the script itself

**Comment by rahsheen** [comment]
What should this reservation be doing? It seems to want to create a Google calendar event somewhere?

**Comment by garritfra** [comment]
@rahsheen yeah, it creates a calendar event for something like a hotel reservation system. this is just a private website project for an appartment i'm planning on renting

**Comment by jgonggrijp** [comment]
Generally, when you want to process the contents of a form with a script on the page, without executing the browser's default behavior of fetching a new page (as a side effect of submitting the data to a remote URL), you need to do something along the following lines.

```javascript
$(function(){
    // this will be run when the document is ready
    // I recommend to give your form an id
    $('#my-form').submit(function(event) {
        event.preventDefault();
        doSomethingWithMyForm();
    }
}
```

You can further improve upon this recipe by using [jQuery.validate](https://jqueryvalidation.org). You don't actually need to set a `target` or an `action` on the form if you only process it with JavaScript.

**Comment by garritfra** [comment]
@jgonggrijp thanks, I'll give it a try. Can this run js code from a seperate file though? The file is fairly long and I don't really want to embed 100 lines of code into my main html

**Comment by jgonggrijp** [comment]
Yes it can. `doSomethingWithMyForm` could be defined in a separate JS file, or even all of the example code that I wrote above. This is possible because when you do

```html
<script src="whatever/name.js"></script>
```

this is functionally equivalent to

```html
<script>
    // contents that you would otherwise put in whatever/name.js
</script>
```

In other words, linked JS files are executed as if they were embedded in the part of the page where you included them.

For very clean ways to work with multiple JS files, you could (optionally) look into things like [Require.js](http://requirejs.org) and [webpack](https://webpack.js.org).
